media/js/base/datalayer-productdownload.es6.js

```diff
@@ -93,40 +106,83 @@ TrackProductDownload.getEventFromUrl = (downloadURL) => {
         // product is first word of product param
         const product = productSplit[0];
         let platform = params.os;
+        // change platform to macos if it's osx
         platform = platform === 'osx' ? 'macos' : platform;
+        // append 'msi' to platform if msi is in the product parameter
+        platform =
+            productParam.indexOf('msi') !== -1 ? platform + '-msi' : platform;
         // release channel is second word of product param
-        // (except for release which says 'latest' but we want 'release')
-        const releaseChannel =
-            productSplit[1] === 'latest' ? 'release' : productSplit[1];
+        let releaseChannel = productSplit[1];
+        // (except for latest, msi, or sub installer - we update those to say release)
+        if (['latest', 'stub', 'msi'].includes(releaseChannel)) {
+            releaseChannel = 'release';
+        }
 
         eventObject = TrackProductDownload.getEventObject(
             product,
             platform,
+            'site',
             releaseChannel,
             params.lang
         );
-    } else if (playStoreURL.test(downloadURL)) {
+    } else if (playStoreURL.test(downloadURL) || marketURL.test(downloadURL)) {
         const idParam = params.id;
-        let androidRelease = 'release';
-        // Android Play Store, need to check release, beta, nightly
+        let androidProduct = 'unrecognized';
+        let androidRelease = '';
+
         switch (idParam) {
+            case 'org.mozilla.firefox':
+                androidProduct = 'firefox';
+                androidRelease = 'release';
+                break;
             case 'org.mozilla.fenix':
+                androidProduct = 'firefox';
                 androidRelease = 'nightly';
                 break;
             case 'org.mozilla.firefox_beta':
+                androidProduct = 'firefox';
                 androidRelease = 'beta';
                 break;
+            case 'org.mozilla.focus':
+                androidProduct = 'focus';
+                break;
+            case 'org.mozilla.klar':
+                androidProduct = 'klar';
+                break;
+            case 'com.ideashower.readitlater.pro':
+                androidProduct = 'pocket';
+                break;
         }
+
         eventObject = TrackProductDownload.getEventObject(
-            'firefox',
+            androidProduct,
             'android',
+            'store',
             androidRelease
         );
-    } else if (appStoreURL.test(downloadURL)) {
+    } else if (appStoreURL.test(downloadURL) || iTunesURL.test(downloadURL)) {
+        let iosProduct = 'unrecognized';
+        if (downloadURL.indexOf('/id989804926') !== -1) {
```

Comment:
10 minutes of exploration tells me this could be complicated because the store links are often included in the adjust links. Can I add this to my list of future improvements and merge this as is? (once I fix the IE compatibility problem)